great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3001,58 +2984,16 @@ def add_profiler(
         Returns:
             The persisted Profiler constructed by the input arguments.
         """
-        return self._add_profiler(
+        return RuleBasedProfiler.add_profiler(
+            data_context=self,
+            profiler_store=self.profiler_store,
```

Comment:
I really want to migrate this logic from static RBP methods to ProfilerStore methods - this will have to be done in a later effort as it is too cumbersome at the moment